src/pudl_archiver/archivers/eia/eia176.py

```diff
@@ -111,6 +128,81 @@ async def get_year_resource(self, year: str, items_list: list[str]) -> ResourceI
             encoding="utf-8",
             index=False,
         )
+        return csv_name, csv_data
+
+    async def get_year_reports_and_custom_resource(
+        self, year: str, datasets_list: list[EIANaturalGasData], items_list: list[str]
+    ) -> ResourceInfo:
+        """Download all available data for a year with multiple reports.
+
+        We do this by constructing the URL based on the EIANaturalGasData object,
+        getting the JSON and transforming it into a csv that is then zipped.
+        We also grab all custom fields from the custom report.
+
+        Args:
+            year: the year we're downloading data for
+            dataset: the report we're downloading
```

Comment:
Docstring/args explanation of how the `datasets_list` and `items_list` work / map to what's getting downloaded would be helpful.